powersimdata/tests/mock_scenario.py

```diff
@@ -34,3 +38,26 @@ def __init__(self, grid_attrs=None, **kwargs):
     def __class__(self):
         """If anyone asks, I'm a Scenario object!"""
         return Scenario
+
+    def __dir__(self):
+        return sorted(super().__dir__() + list(self.state.exported_methods))
+
+    def __getattr__(self, name):
+        if name in self.state.exported_methods:
+            return getattr(self.state, name)
+        elif hasattr(self.state, "__getattr__"):
+            return self.state.__getattr__(name)
+        else:
+            raise AttributeError(
+                f"Scenario object in {self.state.name} state "
+                f"has no attribute {name}"
+            )
+
+    def __setattr__(self, name, value):
+        if name in self._setattr_allowlist:
+            super().__setattr__(name, value)
+        elif name in self.state.exported_methods:
+            raise AttributeError(
+                f"{name} is exported from Scenario.state, edit it there if necessary"
```

Comment:
Scenario.state -> MockScenario.state ?

Reply:
Good question. The intention is the get the user to edit these via `object_name.state.foo`, rather than `object_name.foo`. Since we want this mock to be as transparent as close as possible to the Scenario, I kept this message the same, but if you think it is better to change it I can do that.